stdlib/LinearAlgebra/src/triangular.jl

```diff
@@ -2112,48 +2265,39 @@ end
 unw(x::Real) = 0
 unw(x::Number) = ceil((imag(x) - pi) / (2 * pi))
 
-# End of auxiliary functions for matrix logarithm and matrix power
-
-function sqrt(A::UpperTriangular)
-    realmatrix = false
-    if isreal(A)
-        realmatrix = true
-        for i = 1:checksquare(A)
-            x = real(A[i,i])
-            if x < zero(x)
-                realmatrix = false
-                break
-            end
+# compute A / B for upper quasi-triangular B
+function rdiv_quasitriu(A, B)
+    istriu(B) && return A / UpperTriangular(B)
+    n = checksquare(B)
+    # complete LU decomposition with no pivoting
```

Comment:
Is this approach recommended by Higham? I'd think that using a couple of Givens rotations to annihilate the 2x2 blocks and then use a triangular solve?

Reply:
Higham gave no guidance on this at all and surprisingly I couldn't find anyone else who did either. I don't know much about Givens rotations, hence the use of LU. Could you elaborate on the Givens approach? What would be the advantages over the LU approach used here?

Reply:
See the following calculation. The `U`s are upper triangular and the `T`s are dense.

![2021-03-10 12 53 33](https://user-images.githubusercontent.com/505001/110628390-3db73000-81a3-11eb-8563-03aaa55f8411.jpg)

So essentially something like
```julia
julia> A
4×4 Matrix{Float64}:
  0.151601   -1.01437   2.38495     0.43551
 -0.0627892   0.344165  0.4583      1.27257
 -1.01306    -0.850143  0.811163    0.140954
  0.0458491   1.14956   0.0899956  -1.60257

julia> B
4×4 Matrix{Float64}:
 0.771862  -0.0368294  0.593163  -0.144476
 0.0        1.25168    1.07226   -1.31585
 0.0       -0.0210916  1.06118    0.620557
 0.0        0.0        0.0       -0.740207

julia> G = givens(B[3,3], B[3,2], 2, 3)[1]
LinearAlgebra.Givens{Float64}(2, 3, 0.9998025381561774, -0.01987170592740976)

julia> (A*G)/UpperTriangular(B*G)
4×4 Matrix{Float64}:
  0.19641    -0.755741   2.90131    3.1491
 -0.0813477   0.275915   0.198552  -2.02736
 -1.31249    -0.680985   2.18614    3.10909
  0.0594007   0.905616  -0.863472  -0.180366

julia> A/B
4×4 Matrix{Float64}:
  0.19641    -0.755741   2.90131    3.1491
 -0.0813477   0.275915   0.198552  -2.02736
 -1.31249    -0.680985   2.18614    3.10909
  0.0594007   0.905616  -0.863472  -0.180366
```

Reply:
Thanks! This was simpler than what I was doing and a little bit more efficient.